specification/security/resource-manager/Microsoft.Security/preview/2017-08-01-preview/security.json

```diff
@@ -668,6 +668,267 @@
+                        "schema": {
+                            "$ref": "#/definitions/CloudError"
+                        }
+                    }
+                },
+                "x-ms-pageable": {
+                    "nextLinkName": "nextLink"
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Security/IoTSecuritySolutions/{solutionName}": {
+            "get": {
+                "x-ms-examples": {
+                    "Get an iot security solution": {
+                        "$ref": "./examples/IoTSecuritySolutions/GetIoTSecuritySolution_example.json"
+                    }
+                },
+                "tags": ["IoT Security Solutions"],
+                "description": "Details of a specific iot security solution",
+                "operationId": "IotSecuritySolution_Get",
+                "parameters": [
+                  {
+                      "$ref": "#/parameters/ApiVersion"
+                  },
+                  {
+                      "$ref": "#/parameters/SubscriptionId"
+                  },
+                  {
+                    "$ref": "#/parameters/ResourceGroupName"
+                  },
+                  {
+                      "$ref": "#/parameters/SolutionName"
+                  }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/IoTSecuritySolutionModel"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/CloudError"
+                        }
+                    }
+                }
+            },
+            "put": {
+                "x-ms-examples": {
+                    "Create an iot security solution": {
+                        "$ref": "./examples/IoTSecuritySolutions/CreateIoTSecuritySolution_example.json"
+                    }
+                },
+                "tags": ["IoT Security Solutions"],
+                "description": "Create new solution manager",
+                "operationId": "IotSecuritySolution_Create",
+                "parameters": [
+                  {
+                      "$ref": "#/parameters/ApiVersion"
+                  },
+                  {
+                      "$ref": "#/parameters/SubscriptionId"
+                  },
+                  {
+                    "$ref": "#/parameters/ResourceGroupName"
+                  },
+                  {
+                      "$ref": "#/parameters/SolutionName"
+                  },
+                  {
+                    "$ref": "#/parameters/IotSecuritySolutionData"

```

Comment:
request body and response body should use the same models. This makes it easy for GET-PUT pipeline. The entity returned using GET can easily be used to re-put on the PUT. If there are properties which cant be set on the PUT, mark them as readonly true

Reply:
Its the same model type (`IotSecuritySolutionModel`) - `IotSecuritySolutionData` its the parameter name.
any parameter in this security APIs defines separately at the end of the file.